bin/fetch.rs

```diff
@@ -12,6 +12,7 @@ use alloy_primitives::{Address, B256};
 use alloy_provider::{Provider, ProviderBuilder};
 use alloy_rpc_types::{BlockId, BlockTransactionsKind};
 use clap::Parser;
+use flate2::{bufread, write::GzEncoder, Compression};
```

Comment:
I just realized putting `bufread::GzDecoder` here would be cleaner & more symmetric.